airbyte-webapp/src/packages/cloud/services/dbtCloud.ts

```diff
@@ -0,0 +1,116 @@
+// This module is for the business logic of working with dbt Cloud webhooks.
+// Static config data, urls, functions which wrangle the APIs to manipulate
+// records in ways suited to the UI user workflows--all the implementation
+// details of working with dbtCloud jobs as webhook operations, all goes here.
+// The presentation logic and orchestration in the UI all goes elsewhere.
+//
+// About that business logic:
+// - for now, the code treats "webhook operations" and "dbt Cloud job" as synonymous.
+// - custom domains aren't yet supported
+
+import isEmpty from "lodash/isEmpty";
+
+import { OperatorType, WebBackendConnectionRead, OperationRead } from "core/request/AirbyteClient";
+import { useWebConnectionService } from "hooks/services/useConnectionHook";
+import { useCurrentWorkspace } from "hooks/services/useWorkspace";
+import { useUpdateWorkspace } from "services/workspaces/WorkspacesService";
+
+export interface DbtCloudJob {
+  account: string;
+  job: string;
+  operationId?: string;
+}
+const dbtCloudDomain = "https://cloud.getdbt.com";
+const webhookConfigName = "dbt cloud";
+const executionBody = `{"cause": "airbyte"}`;
+const jobName = (t: DbtCloudJob) => `${t.account}/${t.job}`;
+
+const toDbtCloudJob = (operation: OperationRead): DbtCloudJob => {
+  const { operationId } = operation;
+  const { executionUrl } = operation.operatorConfiguration.webhook || {};
+
+  const matches = (executionUrl || "").match(/\/accounts\/([^/]+)\/jobs\/([^]+)\//);
```

Comment:
For my understanding, does it mean the job actually only stores the full execution URL, while we're entering in the UI individual separate fields?

I think the backend should store the fields we want the user to actually enter, and just build the executionUrl when doing the request. Parsing this client side sounds like a fragile method, and if we don't store this data now, we might have a problem addressing this in a backwards compatible way easily, since we already lost the individual field information.

Reply:
Good callout! We did discuss this a bit.

The backend is actually unaware of dbt accounts/jobs, or dbt cloud at all - it's a "webhook operation", and it just holds the plain URL. dbt Cloud is a totally frontend concept at the moment.

With that said, I do think a better approach would be what you're describing, and the way I'd do it would be to accept the URL "template" that includes named params; plus a map of the params. e.g., `"https://cloud.getdbt.com/api/v2/accounts/{accountId}/jobs/{jobId}/run/", {"accountId": "x", "jobId": "y"}`.

I think for existing configs we would represent this like `"https://cloud.getdbt.com/api/v2/accounts/x/jobs/y/run/", {}`. Since dbt Cloud is going to be the only "webhook operation" integration in the UI for the foreseeable future, once we have the `template, params` representation we can run a targeted migration since we know what these look like, and then we'd be able to kill this bit of messy FE code.

Reply:
@davinchia any thoughts here? I do agree we should make the change in the backend ~asap.

Reply:
I 100% agree; I think there should be cloud endpoints which are specific to dbt Cloud, return the easy-to-use `DbtCloudJob` objects, and, down the line, support UI niceties like selecting dbt jobs from a dropdown (which has to go through the backend for CORS reasons) and the fact that these are backed by a generic webhook record remaining an implementation detail hidden from the frontend code. It came up a bit in earlier discussions and it's something I'm going to advocate for in the immediate clean-up planning/spec work. Thoughts, @mfsiega-airbyte?

For what it's worth, this seems like a reasonable PR sequencing of that change if we do choose to ship the MVP with these API contracts still in place: 
1) first an "append-only" PR to `airbyte-cloud` adding new dbt-Cloud-specific APIs (which wouldn't yet be used); 
2) then full-stack PR to `airbyte` to define and generate those new APIs and replace all the webhook-oriented code in this file with them.

That said, in terms of this PR, I didn't have a working instance of this more minimal backend implementation to develop and experiment with until this weekend, so I had little choice but to run with what was available to get something working for the demo.